custom_resnet_exp.py

```diff
@@ -0,0 +1,200 @@
+import argparse
+from datetime import datetime as dt
+import os
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.preprocessing.image import ImageDataGenerator
+from tensorflow.keras import backend as K
+from lib import custom_resnet
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--size", type=int, default=8,
+                    help="tensor size along X and Y axis")
+parser.add_argument("--aug", type=bool, default=1,
+                    help="whether to use data augmentation (default 1)")
+parser.add_argument("--pixel_mean", type=bool, default=1,
+                    help="whether to subtract the pixel mean (default 1)")
+parser.add_argument("--batch_size", type=int, default=32,
+                    help="batch size (default is 32)")
+parser.add_argument("--num_blocks", type=int, default=16,
+                    help="number of residual blocks")
+parser.add_argument("--dropout", type=float, default=0.2,
+                    help="Dropout (default is 0.2)")
+parser.add_argument("--in_adapter_type",
+                    choices=["original", "space2depth", "strided"],
+                    default="original",
+                    help="Output adapter type (default is original)")
+parser.add_argument("--out_adapter_type",
+                    choices=["v1", "v2", "isometric", "dethwise"],
+                    default="v1",
+                    help="Output adapter type (default is v1)")
+parser.add_argument("--lsmooth",
+                    type=float,
+                    default=0,
+                    help="Label smoothing parameter (default is 0)")
+parser.add_argument("--kernel_reg",
+                    type=float,
+                    default=1e-5,
+                    help="kernel regularization parameter (default is 1e-5)")
+parser.add_argument("--lr",
+                    type=float,
+                    default=1e-3,
+                    help="learning rate (default is 0.01)")
+args = parser.parse_args()
+
+h, w, c = (32, 32, 3)
+num_classes = 10
+
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.cifar10.load_data()
+y_train = tf.keras.utils.to_categorical(y_train, 10)
+y_test = tf.keras.utils.to_categorical(y_test, 10)
+
+# Normalize data.
+x_train = x_train.astype('float32') / 255
+x_test = x_test.astype('float32') / 255
+
+if args.pixel_mean:
+  x_train_mean = np.mean(x_train, axis=0)
+  x_train -= x_train_mean
+  x_test -= x_train_mean
+
+class LRTensorBoard(tf.keras.callbacks.TensorBoard):
+  def __init__(self, log_dir, **kwargs):
+    super().__init__(log_dir=log_dir, **kwargs)
+
+  def on_epoch_end(self, epoch, logs=None):
+    logs = logs or {}
+    logs.update({'lr': K.eval(self.model.optimizer.lr)})
+    super().on_epoch_end(epoch, logs)
+
+def lr_schedule(epoch):
```

Comment:
Experiments share some code (like this function). Makes sense to move it out to shared "util.py" or something along these lines.